### PR TITLE
Convert bunch of files to TS

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

File: src/DecryptionFailureTracker.ts [hunk 1]
```diff
@@ -14,34 +14,40 @@ See the License for the specific language governing permissions and
...
-        this.failedEventId = failedEventId;
-        this.errorCode = errorCode;
+    public readonly ts: number;
+
+    constructor(public readonly failedEventId: string, public readonly errorCode: string) {
         this.ts = Date.now();
     }
 }
 
+type Fn = (count: number, trackedErrCode: string) => void;
```
Comment (turt2live) [inline src/DecryptionFailureTracker.ts:28]:
```suggestion
type TrackingFn = (count: number, trackedErrCode: string) => void;
```

File: src/DecryptionFailureTracker.ts [hunk 2]
```diff
@@ -67,17 +73,14 @@ export class DecryptionFailureTracker {
      * @param {function?} errorCodeMapFn The function used to map error codes to the
      * trackedErrorCode. If not provided, the `.code` of errors will be used.
      */
-    constructor(fn, errorCodeMapFn) {
+    constructor(private readonly fn: Fn, private readonly errorCodeMapFn?: ErrCodeMapFn) {
```
Comment (turt2live) [inline src/DecryptionFailureTracker.ts:76]:
```suggestion
    constructor(private readonly fn: TrackingFn, private readonly errorCodeMapFn?: ErrCodeMapFn) {
```

File: src/HtmlUtils.tsx [hunk 3]
```diff
@@ -523,7 +527,7 @@ export function linkifyElement(element: HTMLElement, options = linkifyMatrix.opt
  * @param {object} [options] Options for linkifyString. Default: linkifyMatrix.options
  * @returns {string}
  */
```
Comment (turt2live) [inline src/HtmlUtils.tsx:529]: the linkifyString/Element functions above could do with return types while you're in the area.

maybe some brief doc fixes too...

File: src/utils/EditorStateTransfer.ts [hunk 4]
```diff
@@ -14,36 +14,40 @@ See the License for the specific language governing permissions and
...
-    }
+    private serializedParts: SerializedPart[] = null;
+    private caret: Caret = null;
+
+    constructor(private readonly event: MatrixEvent) {}
 
-    setEditorState(caret, serializedParts) {
-        this._caret = caret;
-        this._serializedParts = serializedParts;
+    setEditorState(caret: Caret, serializedParts: SerializedPart[]) {
```
Comment (turt2live) [inline src/utils/EditorStateTransfer.ts:33]: explicit member access on all of these

**Review by turt2live — COMMENTED** [review]
otherwise looks good

**Review by turt2live — APPROVED** [review]

**Comment by t3chguy** [comment]
E2E test itself passed, benchmark gitting failed

**Pull request merged**